### PR TITLE
Fixes transformer

### PR DESCRIPTION
**Pull request by rPraml**
Here are some suggestions what I've changed in the transformer. This fixes https://groups.google.com/forum/#!topic/ebean/asmX4khWNmE

### CODE REVIEW

File: src/main/java/io/ebean/enhance/Transformer.java [hunk 1]
```diff
@@ -37,6 +40,26 @@
...
+  static {
+    try {
+      Properties prop = new Properties();
+      InputStream in = Transformer.class.getResourceAsStream("/META-INF/maven/io.ebean/ebean-agent/pom.properties");
+      if (in != null) {
+        prop.load(in);
+        in.close();
+        version = prop.getProperty("version");
+      }
+      System.out.println("ebean-agent version: " + version);
```
Comment (rPraml) [inline src/main/java/io/ebean/enhance/Transformer.java]: I find it useful, if the agent outputs it's version on first load

**Comment by rbygrave** [comment]
Hmm, struggling to read this ... not sure what is being changed by the try catch etc.  I'll have to put this on hold until I have time to look at what is trying to be achieved here.

File: src/main/java/io/ebean/enhance/Transformer.java [hunk 2]
```diff
@@ -110,42 +133,46 @@ public int getLogLevel() {
...
       if (isEbeanModel || enhanceContext.detectEntityTransactionalEnhancement(className)) {
-
-        DetectEnhancement detect = detect(loader, classfileBuffer);
-
-        if (detect.isEntity()) {
-          if (detect.isEnhancedEntity()) {
-            detect.log(3, "already enhanced entity");
-          } else {
-            entityEnhancement(loader, request);
+        try {
```
Comment (rbygrave) [inline src/main/java/io/ebean/enhance/Transformer.java]: It is not clear to me what this try { ... is here.  What problem is this solving ?

File: src/main/java/io/ebean/enhance/ant/OfflineFileTransform.java [hunk 3]
```diff
@@ -150,7 +153,7 @@ private void transformFile(File file) throws IOException, IllegalClassFormatExce
 
 		if (result != null) {
 			InputStreamTransform.writeBytes(result, file);
-			if(listener!=null) {
+			if(listener!=null && logLevel > 0) {
```
Comment (rbygrave) [inline src/main/java/io/ebean/enhance/ant/OfflineFileTransform.java]: Is this useful?  What is the motivation for this?  Controlling maven logging?

Comment (rPraml) [reply src/main/java/io/ebean/enhance/ant/OfflineFileTransform.java]: The concrete problem was, that the OfflineFileTransformer flooded the build-log.

Motivation:
I use the offline-filetransformer in a self compiled version of the "plexus-compiler-eclipse"
the modified version supports annotation-processing for lombok & querybean.
It performs compilation and enhancement in one step. (and due the fact that the eclipse-compiler supports multi-threading, we could nearly half our maven build tim)

File: src/main/java/io/ebean/enhance/Transformer.java (hunk 2 above)
Comment (rPraml) [reply src/main/java/io/ebean/enhance/Transformer.java]: When a "NoEnhancementRequiredException" is thrown during "detect", "entiyEnhancement" or "transactionalEnhancement" - the querybean-enhancement was skipped.

Unfortunately, I really don't remember under which constellation this occured here (but we've often the problem, that querybeans were not properly enhanced)

**Pull request closed without merging**